spec/breadbox/s3_client_spec.rb

```diff
@@ -91,6 +91,24 @@ module Breadbox
                                                      .with(file, options)
         client.upload(path: "/", file: file, content_type: "image/jpeg")
       end
+
+      it "passes path parameter" do
+        file    = File.open("./tmp/new-file.jpg")
+        bucket_object = client.s3_bucket_object.objects["new-file.jpg"]
+        expect_any_instance_of(AWS::S3::ObjectCollection).to receive(:[])
+          .with("my-path/new-file.jpg") { bucket_object }
+        expect_any_instance_of(AWS::S3::S3Object).to receive(:write)
+        client.upload(path: "my-path", file: file, content_type: "image/jpeg")
+      end
+
+      it "passes filename parameter" do
+        file    = File.open("./tmp/new-file.jpg")
+        bucket_object = client.s3_bucket_object.objects["new-file-2.jpg"]
+        expect_any_instance_of(AWS::S3::ObjectCollection).to receive(:[])
+          .with("new-file-2.jpg") { bucket_object }
```

Comment:
This is probably my own fault for using this syntax somewhere else --> but I feel like the `{ }` syntax instead of saying `.and_return` is confusing.